### DISCUSSION TITLE
How to display more detailed log output when npm run build?

### DISCUSSION THREAD

**Discussion by SageJustus** [vercel/next.js#65402] [Help]
### Summary

When I run npm run build, I'm stuck at the `Collecting page data` step.

```
>yarn run build
yarn run v1.22.22
$ next build --debug --no-lint
 ⚠ Linting is disabled.
  ▲ Next.js 14.2.3
  - Environments: .env.production

   Creating an optimized production build ...
 ✓ Compiled successfully
   Skipping validation of types
 ✓ Collecting page data
```

What bothers me is that the console doesn't output any detailed logs and I don't know where the error is occurring.

How can I display more detailed logs? I've used the `next build --debug` script but this has no effect.

**Comment by khludenevav** [comment]
Same question.

**Comment by andreymaklakov** [comment]
Any updates?

**Comment by Sljcom** [comment]
Any updates?

**Comment by JoeyKhd** [comment]
Any updates?

```
> next build "-d"

▲ Next.js 15.3.2

Creating an optimized production build ...
uncaughtException [TypeError: Cannot read properties of undefined (reading 'length')]
```

Trying to find out where the error is being thrown, but even with -d or --debug nothing is really shown

**Comment by jlewin** [comment]
I too was pulling my hair out with failing builds on `Next.js 15.2.4`, far too much build text output, and this gem of an error:

> TypeError: Cannot read properties of undefined (reading 'length')

After failing to find any valid fixes and being stuck unable to build I decided to kill my `.next` and `node_modules` directories and `pnpm install` may way back to a fresh env. After a brute force purge, I'm again able to build without error. Not sure if it's a reasonable fix but figured I'd share my success.

**Comment by kev202209** [comment]
I too got stuck but with a different error:

> Build error occurred
[TypeError: Expected a non-empty string]
 ELIFECYCLE  Command failed with exit code 1.

Does anyone know how to fix this? I have already tried to make a lot of changes that didn't lead to a solution, I even removed the .next and node_modules and reinstalled, but the error persists.
